### PR TITLE
Add date time picker #LMR-1458 #LMR-1522

### PR DESCRIPTION
**Pull request by livthomas**
* respect locale in date picker
* fix long text in constraint config modal title
* extract valid data colors to new file

### CODE REVIEW

**Comment by livthomas** [comment]
If you set date time constraint with just year format (e.g. `YYYY`), you can see a strange behavior of the date picker because of a [bug](https://github.com/valor-software/ngx-bootstrap/issues/5154) in `ngx-bootstrap` library.

File: src/app/shared/date-time/picker/date-time-picker.component.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    // otherwise it is immediately closed in table
+    event.stopPropagation();
+  }
+
+  public open() {
+    if (this.overlayRef) {
+      return;
+    }
+
+    this.overlayRef = this.overlay.create({
```
Comment (kubedo8) [inline src/app/shared/date-time/picker/date-time-picker.component.ts]: Interesting! Can you tell me something about it on todays meeting?

File: src/app/shared/date-time/input/date-time-input.component.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  public id: string;
+
+  @Input()
+  public min: Date;
+
+  @Input()
+  public max: Date;
+
+  @Input()
+  public origin: ElementRef | HTMLElement;
```
[the remark this answers is not in the document]
Comment (livthomas) [reply src/app/shared/date-time/input/date-time-input.component.ts]: Thanks. I forgot to remove it after the refactoring.

File: src/app/shared/date-time/picker/date-time-picker.component.ts (hunk 1 above)
Comment (livthomas) [reply src/app/shared/date-time/picker/date-time-picker.component.ts]: Yes, but it is just normal [Overlay](https://material.angular.io/cdk/overlay/overview) from Angular CDK.

**Pull request merged**